regen/HeaderParser.pm

```diff
@@ -659,6 +659,13 @@ sub parse_fh {
                 $type= "content";
                 $sub_type= "#error";
             }
+            elsif ($flat =~ /#\s*\z/) {
+                # deal with the unusual case of # /* comment */ or a plain # on a line
+                # technically this is not legal, but we have had one in scope.h for
```

Comment:
Why is it not legal? See e.g. https://en.cppreference.com/w/c/preprocessor or https://stackoverflow.com/questions/35207515/what-is-the-purpose-of-a-single-pound-hash-sign-on-its-own-line-in-the-c-c.

Reply:
Fixed.